test/boost/managed_bytes_test.cc

```diff
@@ -98,6 +98,24 @@ BOOST_AUTO_TEST_CASE(test_copy_constructor) {
     }
 }
 
+// Reproducer for #23781.
+// Simulates a copy of a large cell from the standard allocator to lsa.
+BOOST_AUTO_TEST_CASE(test_copy_constructor_standard_to_lsa) {
+    fragmenting_allocation_strategy alloc_1(128 * 1024);
+    fragmenting_allocation_strategy alloc_2(128 * 1024 / 10);
+
+    for (size_t size : {16*1024, 32*1024, 64*1024, 128*1024, 256*1024}) {
+        auto b = tests::random::get_bytes(size);
+        with_allocator(alloc_1, [&] {
+            auto m = managed_bytes(b);
+            with_allocator(alloc_2, [&] {
+                auto m_copy_different_allocator = m;
+                BOOST_CHECK_EQUAL(m, m_copy_different_allocator);
```

Comment:
Can we check the fragment sizes too? I suppose they are maybe hidden in the private layers.
Maybe a special allocator which creates stats on allocations, so you can check what allocations were used in the copy?

Reply:
`fragmenting_allocation_strategy` (used by all the tests in this file) already has `BOOST_CHECK_LE(size, _preferred_max_contiguous_allocation)`.

Reply:
Unless you meant that you want a stronger check that all non-last fragments have size *equal* to `_preferred_max_contiguous_allocation`. Then yes, this is something that could be added to the test, but I don't know if we want to do this right now, and risk delaying the bugfix.

Reply:
I think I'm happy with the current test, as you explained it. On first glance (now knowing about the checks in fragmenting allocator) it seemed to not test the fragmenting.